### PR TITLE
All parameters as inputs and use of 0.4 version

### LINKED ISSUE #8: Update app to newest smart contracts
Use new npm package 0.4.0

This entails:
- sending over auctionEndDate , instead of auctionDuration to the contract

- sending over orderCancelationEndDate , instead of orderCancelationDuration to the contract

- sending over one more field: [`accessManagerContractData`](https://github.com/gnosis/ido-contracts/blob/main/contracts/EasyAuction.sol#L163)
Initially, for the first use cases, accessManagerContractData should be just containing an address. It should be send over as bytes like [that](https://github.com/gnosis/ido-contracts/blob/main/test/contract/AllowListManager.spec.ts#L49):

### LINKED ISSUE #9: Make all paramters of initiateAuction as form - inputs
All fields from the initiateAuction scripts should become form input values with validations:
```
.addParam(
      "auctioningToken",
      "The ERC20's address of the token that should be sold",
    )
    .addParam(
      "biddingToken",
      "The ERC20's address of the token that should be bought",
    )
    .addParam(
      "sellAmount",
      "The amount of auctioningTokens to be sold in atoms",
    )
    .addParam(
      "minBuyAmount",
      "The amount of biddingToken to be bought at least for selling sellAmount in atoms",
    )
    .addOptionalParam(
      "minFundingThreshold",
      "The minimal funding threshold for executing the settlement. If funding is not reached, everyone will get back their investment",
      "0",
      types.string,
    )
    .addOptionalParam(
      "orderCancellationEndDate",
      "The timestamp (in seconds) until which orders can be canceled",
      undefined,
      types.string,
    )
    .addOptionalParam(
      "auctionEndDate",
      "The timestamp (in seconds) marking the end of the auction",
      undefined,
      types.string,
    )
    .addOptionalParam(
      "minBuyAmountPerOrder",
      "Describes the minimal buyAmount per order placed in the auction. This can be used in order to protect against too high gas costs for the settlement",
      "0.01",
      types.string,
    )
    .addOptionalParam(
      "isAtomicClosureAllowed",
      "Describes whether the auction should be allowed to be closed atomically",
      false,
      types.boolean,
    )
    .addOptionalParam(
      "allowListManager",
      "Contract address for a potential allowListManger contract, if allow listing is wanted for the started auction",
      "0x0000000000000000000000000000000000000000",
      types.string,
    )
    .addOptionalParam(
      "allowListData",
      "provide data that is needed for the allowListManager",
      "0x",
      types.string,
    )
    
```

**Valdiations:**
allowListData -> check whether its a valid EOA accounts
allowListManager -> check that its a deployed contract on the network
Check that orderCancellationEndDate<auctionEndDate
Check that now<auctionEndDate

Additional information:
Please provide for each input field some kind of tooltip to provide the information from the description of the initiateAuction hardhat script.

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #8 
Closes #9

This still needs some work, I think is better typed and organized. 

- [x] Some validations aren't working properly, like the one that says 'Invalid ERC20'. Could be related to `useIsContract` 
- [x] Other validations run after submit is clicked, this is easier for async validations.
- [ ] Still pending a modal with the Id of the auction.

### CODE REVIEW

**Comment by josojo** [comment]
Great, nice start.

Some immediate improvements for usability:
- AllowlistingContract must be optional.
- Data for allowlistmanager could be called signer and we just want to provide a only an address:
It was very cryptically in the requirements here:

>Initially, for the first use cases, accessManagerContractData should be just containing an address. It should be sent over as bytes like that:

**Comment by josojo** [comment]
progress, yeah!

> I can not yet place orders:

image

Could we name this field: AllowList Contract?
image

**Comment by hpmaxi** [comment]
@josojo Some errors should appear now.  I'm maintaining a list of issues in #12 . Also changed the label that you asked for.

File: src/hooks/useSubmitAuction.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    }
+
+    const valuesToSend: ValuesToSend = [
+      auctioningToken.address,
+      biddingToken.address,
+      moment(orderCancellationEndDate).unix().toString(),
+      moment(auctionEndDate).unix().toString(),
+      sellAmountInAtoms,
+      minBuyAmountInAtoms,
+      minimumBiddingAmountPerOrder,
```
Comment (josojo) [inline src/hooks/useSubmitAuction.tsx]: I think this one should also be in atoms. Maybe I am even doing that wrong in the hardhat scripts.

File: src/formConfig.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      'The amount of bidding token to be bought at least for selling the amount to sell in atoms',
+  },
+  minFundingThreshold: {
+    label: 'Minimal Funding for executing the settlement',
+    tooltipText:
+      'The minimal funding threshold for executing the settlement. If funding is not reached, everyone will get back their investment',
+  },
+  orderCancellationEndDate: {
+    label: 'End time for order cancellation',
+    tooltipText: 'The timestamp (in seconds) until which orders can be canceled',
```
Comment (josojo) [inline src/formConfig.ts]: Given that we have the time pickers, I think we should write:

toolttipText: 'Choose a time (local) until users are still able to cancel orders'

File: src/formConfig.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  },
+  auctionEndDate: {
+    label: 'Auction end time',
+    tooltipText: 'The timestamp (in seconds) marking the end of the auction',
```
Comment (josojo) [inline src/formConfig.ts]: toolttipText: 'Choose a time (local) marking the end of the auction.'

File: src/hooks/useSubmitAuction.tsx [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+      minFundingThresholdInAtoms,
+      !!isAtomicClosureAllowed,
+      useDefaultAllowListManager ? '0x0000000000000000000000000000000000000000' : allowListManager,
+      useDefaultAllowListData
+        ? formatBytes32String('0x')
```
Comment (josojo) [inline src/hooks/useSubmitAuction.tsx]: formatBytes32String('0x')

is returning for me: 
```
"0x3078000000000000000000000000000000000000000000000000000000000000"
```

Sending over just '0x' should be fine. It does not need to be bytes32, if we don't have an allowlist manager contract specified

**Comment by josojo** [comment]
Is this PR working for you @hpmaxi ? Somehow, I always get failing txs.

Could you please try to get a minimal version running ? We want to do simulations before tthe tests with boson.

File: src/formConfig.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+export const DEFAULT_FORM_PARAMS: Readonly<Record<FormKeys, any>> = {
+  auctioningToken: '',
+  biddingToken: '',
+  sellAmount: '0',
+  minBuyAmount: '0',
+  minFundingThreshold: '0',
+  orderCancellationEndDate: '',
+  auctionEndDate: '',
+  minBuyAmountPerOrder: '0.01',
+  isAtomicClosureAllowed: true,
```
Comment (josojo) [inline src/formConfig.ts]: Could we please set this value to false?

**Review by josojo — APPROVED** [review]
Nice, I tested it quite a bit and it works. Also, the whitelisting feature works 

I had small crashes during the usage. I put in a high prio issue for that. Thanks

File: src/formConfig.ts (hunk 5 above)
Comment (hpmaxi) [reply src/formConfig.ts]: Done

**Pull request merged**